[2025-08-25T03:00:02.398+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-08-25T03:00:02.413+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: books_pipeline_dag.scrape_books scheduled__2025-08-24T03:00:00+00:00 [queued]>
[2025-08-25T03:00:02.419+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: books_pipeline_dag.scrape_books scheduled__2025-08-24T03:00:00+00:00 [queued]>
[2025-08-25T03:00:02.419+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-08-25T03:00:02.426+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): scrape_books> on 2025-08-24 03:00:00+00:00
[2025-08-25T03:00:02.430+0000] {standard_task_runner.py:64} INFO - Started process 67120 to run task
[2025-08-25T03:00:02.432+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'books_pipeline_dag', 'scrape_books', 'scheduled__2025-08-24T03:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/books_pipeline_dag.py', '--cfg-path', '/tmp/tmp5_kq7bqv']
[2025-08-25T03:00:02.434+0000] {standard_task_runner.py:91} INFO - Job 6: Subtask scrape_books
[2025-08-25T03:00:02.474+0000] {task_command.py:426} INFO - Running <TaskInstance: books_pipeline_dag.scrape_books scheduled__2025-08-24T03:00:00+00:00 [running]> on host 0b591e383dbb
[2025-08-25T03:00:02.532+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='books_pipeline_dag' AIRFLOW_CTX_TASK_ID='scrape_books' AIRFLOW_CTX_EXECUTION_DATE='2025-08-24T03:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-08-24T03:00:00+00:00'
[2025-08-25T03:00:02.533+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-08-25T03:00:04.521+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-08-25T03:00:04.524+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scraper/books_scraper.py", line 91, in scrape_books_to_parquet
    df = scrape_books(category_url)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scraper/books_scraper.py", line 73, in scrape_books
    rows, next_url = _parse_page(html, cur)
                     ^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scraper/books_scraper.py", line 42, in _parse_page
    price = _parse_price(art.select_one("p.price_color").text)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scraper/books_scraper.py", line 27, in _parse_price
    return float(pounds_text.replace("£", "").strip())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: could not convert string to float: 'Â42.96'
[2025-08-25T03:00:04.537+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=books_pipeline_dag, task_id=scrape_books, run_id=scheduled__2025-08-24T03:00:00+00:00, execution_date=20250824T030000, start_date=20250825T030002, end_date=20250825T030004
[2025-08-25T03:00:04.544+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 6 for task scrape_books (could not convert string to float: 'Â42.96'; 67120)
[2025-08-25T03:00:04.590+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-08-25T03:00:04.625+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-08-25T03:00:04.626+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
